src/main/java/tds/testpackageconverter/converter/mappers/TestPackageBlueprintMapper.java

```diff
@@ -24,6 +39,19 @@
      */
     public static List<BlueprintElement> mapBlueprint(final String testPackageName,
                                                       final List<Testspecification> testSpecifications) {
+
+        if (testSpecifications.isEmpty()) {
+            throw new IllegalArgumentException("No test test packages found. Aborting...");
+        }
+
+        // Need to find subject and grade, both of which we assume to be consistent over the set of
+        // test specifications. Therefore, we just examine the first one.
+        Testspecification firstSpecification = testSpecifications.get(0);
```

Comment:
Is there a reason we are getting the subject and grade in this method rather than inside the `mapClaimsAndTargetBlueprintElements` method?  Does the `adminTestPackages` collection not have the data since it is a filtered list (line 67) of the original `TestSpecification` elements?  It'd remove the extra argument to the `mapClaimsAndTargetBlueprintElements`.